codex/blockexchange/engine/engine.nim

```diff
@@ -280,6 +269,20 @@ proc scheduleTasks(b: BlockExcEngine, blocksDelivery: seq[BlockDelivery]) {.asyn
 
           break # do next peer
 
+proc issueCancellations(b: BlockExcEngine, addrs: seq[BlockAddress]) {.async.} =
+  ## Tells neighboring peers that we're no longer interested in a block.
+  trace "Sending block request cancellations to peers", addrs = addrs.len
+
+  let sends = b.peers.mapIt(
+    (peer: it, request: b.network.request.sendWantCancellations(peer = it.id,
+      addresses = addrs)))
+
+  discard await allFinished(sends.mapIt(it.request))
```

Comment:
Please don't use `discard` in situations like this, either use `allFutures`, which will return `void` or use `allFinished` and check for failed futures. 

In general, It's better to always check if some of the futures failed and do something with it - at the very least log it.

Reply:
OK good, I was not aware of the proper way to do this. `allFutures`it is, then.